plugins/@grouparoo/pipedrive/src/lib/export-organizations/destinationMappingOptions.ts

```diff
@@ -0,0 +1,169 @@
+import {
+  DestinationMappingOptionsMethod,
+  DestinationMappingOptionsResponseType,
+  SimpleAppOptions,
+  objectCache,
+} from "@grouparoo/core";
+
+import { connect } from "../connect";
+import { PipedriveClient } from "../client";
+import { GROUP_FIELD_PREFIX } from "./listMethods";
+
+export interface PipedriveCacheData {
+  appId: string;
+  appOptions: SimpleAppOptions;
+}
+
+export const destinationMappingOptions: DestinationMappingOptionsMethod =
+  async ({ appOptions, appId }) => {
+    const client = await connect(appOptions);
+    const allKnown = await getKnownOrganizationFields(
+      client,
+      { appId, appOptions },
+      true
+    );
+
+    const required = allKnown.filter((field) => field.required);
+
+    // Filter out required fields and those that we create to manage group membership
+    const knownNotRequired = allKnown.filter(
+      (field) => !field.required && !field.groupMembershipField
+    );
+
+    return {
+      labels: {
+        property: {
+          singular: "Pipedrive Field",
+          plural: "Pipedrive Fields",
+        },
+        group: {
+          singular: "Pipedrive Filter",
+          plural: "Pipedrive Filters",
+        },
+      },
+      properties: {
+        known: knownNotRequired,
+        required,
+        allowOptionalFromProperties: false,
+      },
+    };
+  };
+
+const mapTypesToGrouparoo = (
+  pipedriveType: string,
+  fieldName: string
+): DestinationMappingOptionsResponseType => {
+  // https://pipedrive.readme.io/docs/core-api-concepts-custom-fields#section-types-of-custom-fields
+
+  const overrides: Record<string, DestinationMappingOptionsResponseType> = {
+    Name: "string",
+  };
+
+  if (overrides[fieldName]) return overrides[fieldName];
+
+  const map: Record<string, DestinationMappingOptionsResponseType> = {
```

Comment:
If feels like this an other methods throughout should be shared between persons and organizations. Can you look into that and recommend the right abstraction level?